tests/test_importer.py

```diff
@@ -69,39 +74,118 @@ def test_number_of_imported_entities(entity_name, Model, data_import, data_folde
+    '''
+    In https://github.com/security-force-monitor/sfm-cms/issues/637, we
+    discovered a bug in which sources were multiplied when derived from the
+    original, mutable source list, rather than a shallow copy. Test data was
+    created such that we could test sources weren't inadvertantly assigned
+    to unrelated attributes. This test asserts that extraneous sources are not
+    created, based on that data.
+
+    N.b., there are not constraints on what data points can share sources
+    outside of this test.
+    '''
+    sourced_attributes = [
+        attr for attr in dir(AccessPoint.objects.first())
+        if attr.endswith('_related')
+    ]
+
+    access_points_for_test = Q()
+
+    for substring in ('alpha', 'beta', 'gamma', 'delta', 'is-member', 'has-member'):
+        access_points_for_test |= Q(source__title__icontains=substring)
+
+    for access_point in AccessPoint.objects.filter(access_points_for_test):
         related_objects = []
-        for attr in src_related_attrs:
-            related_objects += [obj for obj in getattr(access_point, attr).all()
-                                if obj]
+
+        for attr in sourced_attributes:
+            related_objects += [
+                obj for obj in getattr(access_point, attr).all() if obj
+            ]
+
         related_obj_types = set(obj._meta.object_name for obj in related_objects)
+
         if len(related_obj_types) > 1:
-            # Object types that we expect may share sources
+            # Data points can be evidenced directly through an accompanying
+            # source field and indirectly through a related field, e.g., if
+            # Unit A is the parent of Unit B, the source for Unit A's name is
+            # also added to that composition.
+            #
+            # These sets are groupings of attributes that we expect could share
+            # sources based on on our practice of direct and indirect sourcing.
+            # The success of this test depends on the particular assignment of
+            # sources in the test data. If the test data or data model changes,
+            # these sets may need to be adjusted.
             permitted_org_set = set([
-                'CompositionChild', 'OrganizationName',
-                'MembershipOrganizationMember', 'MembershipOrganizationOrganization',
-                'MembershipPersonOrganization', 'MembershipPersonMember',
-                'CompositionParent', 'EmplacementSite', 'EmplacementStartDate',
-                'EmplacementOrganization'
+               'CompositionChild',
+               'CompositionParent',
+               'EmplacementOrganization',
+               'EmplacementSite',
+               'EmplacementStartDate',
+               'MembershipOrganizationMember',
+               'MembershipOrganizationOrganization',
+               'MembershipPersonMember',
+               'MembershipPersonOrganization',
+               'OrganizationName',
+            ])
```

Comment:
None of these sets changed – I just put each item on its own line and sorted them alphabetically, for easier reading.